core/services/bridget/bridget.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from typing import Dict, List
+
+from bridges.bridges import Bridge
+from bridges.serialhelper import Baudrate
+from pydantic import BaseModel, conint
+from serial.tools.list_ports_linux import SysFS, comports
+
+
+class BridgeSpec(BaseModel):
+    """Basic interface for 'bridges' links."""
+
+    serial_path: str
+    baud: Baudrate
+    ip: str
+    udp_port: conint(gt=1023, lt=65536)  # type: ignore
+
+    def __str__(self) -> str:
+        return f"{self.serial_path}:{self.baud}//{self.ip}:{self.udp_port}"
+
+    def __hash__(self) -> int:
+        return hash(str(self))
+
+
+class Bridget:
+    """Manager for 'bridges' links."""
+
+    def __init__(self) -> None:
+        self._bridges: Dict[BridgeSpec, Bridge] = {}
+
+    @staticmethod
+    def is_port_available(port: str) -> bool:
+        try:
+            with open(port, mode="r", encoding="utf-8"):
+                pass
+            return True
+        except Exception:
+            return False
+
+    @staticmethod
+    def available_serial_ports() -> List[str]:
+        available_ports = []
+        for port in comports():
+            if not Bridget.is_port_available(port.device):
```

Comment:
We should also filter ports that are already in use by bridges

![image](https://user-images.githubusercontent.com/1215497/135917722-b468fb8f-e2c5-44f0-82b7-6523b5051c7d.png)

Reply:
Fair